app/controllers/partners_controller.rb

```diff
@@ -106,6 +106,26 @@ def invite
     end
   end
 
+  def invite_and_approve
+    partner = current_organization.partners.find(params[:id])
+
+    svc = PartnerInviteService.new(partner: partner)
+    svc.call
+
+    if svc.errors.none?
+      svc = PartnerApprovalService.new(partner: partner)
```

Comment:
Hmm... yeah, I'd rather have a new service that does this (even if all the service does is call the other 2 services and collate their errors). This is a bit too hefty for a controller.